### PR TITLE
ramips: update mtd configuration for Linksys EA7300 and EA7500

### PR DESCRIPTION
**Pull request by arrmo**
A few minor adjustments to the dtsi file, to enable dual partition functionality
- u_env (U-Boot environment), make r/w
- alternative kernel and rootfs, make r/w
- align partition naming with other dual partition routers (e.g. Linksys EA3500)

Signed-off-by: Russell Morris <rmorris@rkmorris.us>

### CODE REVIEW

File: target/linux/ramips/mt7621/base-files/lib/upgrade/linksys.sh [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		fw_setenv -s - <<-EOF
+			boot_part 1
+			bootcmd "run nandboot"
+		EOF
+		printf "kernel1"
+		return
+		;;
+	*)
+		return
+		;;
```
Comment (adschm) [inline target/linux/ramips/mt7621/base-files/lib/upgrade/linksys.sh:43]: This default case is useless and should be removed.

File: target/linux/ramips/mt7621/base-files/lib/upgrade/linksys.sh [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	fi
+
+	case $cur_boot_part in
+	1)
+		fw_setenv -s - <<-EOF
+			boot_part 2
+			bootcmd "run altnandboot"
+		EOF
+		printf "kernel2"
+		return
```
Comment (adschm) [inline target/linux/ramips/mt7621/base-files/lib/upgrade/linksys.sh:31]: There is no code after this, so return can be dropped here and for 2)

File: target/linux/ramips/mt7621/base-files/lib/upgrade/linksys.sh [hunk 3]
```diff
@@ -0,0 +1,93 @@
+#
+# Copyright (C) 2014 OpenWrt.org
```
Comment (adschm) [inline target/linux/ramips/mt7621/base-files/lib/upgrade/linksys.sh:2]: You are not openwrt. But please add an SPDX license identifier instead.

**Comment by adschm** [comment]
> ramips: update mtd configuration for Linksys EA7300 v2

This is not just affecting EA7300 v2 ...

File: target/linux/ramips/mt7621/base-files/lib/upgrade/linksys.sh (hunk 3 above)
Comment (arrmo) [reply target/linux/ramips/mt7621/base-files/lib/upgrade/linksys.sh:2]: Good point, done!

File: target/linux/ramips/mt7621/base-files/lib/upgrade/linksys.sh (hunk 1 above)
Comment (arrmo) [reply target/linux/ramips/mt7621/base-files/lib/upgrade/linksys.sh:43]: Sure, NP. I admit, copied linksys.sh from another target - but agree with you, updating it!

File: target/linux/ramips/mt7621/base-files/lib/upgrade/linksys.sh (hunk 2 above)
Comment (arrmo) [reply target/linux/ramips/mt7621/base-files/lib/upgrade/linksys.sh:31]: Yep, agreed. Done.

**Comment by arrmo** [comment]
> > ramips: update mtd configuration for Linksys EA7300 v2
> 
> This is not just affecting EA7300 v2 ...

Agreed! Updated the commit message to capture this. Thanks.

**Comment by thosmos** [comment]
What's needed for this to move forward?

I'm attempting to add support for these routers to Althea.net and this dual partition functionality is important so we can handle these similarly to other openwrt dual partition routers.

**Comment by adschm** [comment]
> What's needed for this to move forward?

It's not up-to-date and in Draft state. So, actually there is no reason why anybody serious would have a closer look. (Which does not necessarily mean that you will have much more luck after that ...)

**Pull request closed without merging**